File: Easy/1592. Rearrange Spaces Between Words/1592. Rearrange Spaces Between Words.ts
function reorderSpaces(text: string): string {
    const words: string[] = text.trim().split(/\s+/);
    const wordCount = words.length;
    const spaceCount = (text.match(/ /g) || []).length;

    if (wordCount === 1) {
        return words[0] + ' '.repeat(spaceCount);
    }

    const spacesBetween = Math.floor(spaceCount / (wordCount - 1));
    const extraSpaces = spaceCount % (wordCount - 1);
    const spacesBetweenStr = ' '.repeat(spacesBetween);

    let result = '';
    for (let i = 0; i < wordCount; i++) {
        result += words[i];
        if (i < wordCount - 1) {
            result += spacesBetweenStr;
        }
    }

    result += ' '.repeat(extraSpaces);

    return result;
}

File: Easy/1592. Rearrange Spaces Between Words/1592. Rearrange Spaces Between Words.java
class Solution {
    public String reorderSpaces(String text) {
        String[] words = text.trim().split("\\s+");
        int wordCount = words.length;
        int spaceCount = text.length() - text.replaceAll(" ", "").length();
        
        if (wordCount == 1) {
            return words[0] + " ".repeat(spaceCount);
        }
        
        int spacesBetween = spaceCount / (wordCount - 1);
        int extraSpaces = spaceCount % (wordCount - 1);
        String spacesBetweenStr = " ".repeat(spacesBetween);
        
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < wordCount; i++) {
            result.append(words[i]);
            if (i < wordCount - 1) {
                result.append(spacesBetweenStr);
            }
        }
        
        result.append(" ".repeat(extraSpaces));
        
        return result.toString();
    }
}
